app/models/location.rb

```diff
@@ -398,6 +398,10 @@ def active?
     status == :active
   end
 
+  def deactivated?
+    status == :deactivated
```

Comment:
We should memoize status — otherwise you are executing up to 3 queries per `deactivated?` and so on.